opentelemetry-sdk/src/opentelemetry/sdk/metrics/_internal/aggregation.py

```diff
@@ -433,84 +444,76 @@ def collect(
         """
         Atomically return a point for the current value of the metric.
         """
-        with self._lock:
-            if not any(self._bucket_counts):
-                return None
 
-            bucket_counts = self._bucket_counts
-            start_time_unix_nano = self._start_time_unix_nano
+        with self._lock:
+            current_value = self._current_value
             sum_ = self._sum
-            max_ = self._max
             min_ = self._min
+            max_ = self._max
 
-            self._bucket_counts = self._get_empty_bucket_counts()
-            self._start_time_unix_nano = collection_start_nano
+            self._current_value = None
             self._sum = 0
             self._min = inf
             self._max = -inf
 
-        current_point = HistogramDataPoint(
-            attributes=self._attributes,
-            start_time_unix_nano=start_time_unix_nano,
-            time_unix_nano=collection_start_nano,
-            count=sum(bucket_counts),
-            sum=sum_,
-            bucket_counts=tuple(bucket_counts),
-            explicit_bounds=self._boundaries,
-            min=min_,
-            max=max_,
-        )
+            if (
+                self._instrument_aggregation_temporality
+                is AggregationTemporality.DELTA
+            ):
+                # This happens when the corresponding instrument for this
+                # aggregation is synchronous.
+                if (
+                    collection_aggregation_temporality
+                    is AggregationTemporality.DELTA
+                ):
 
-        if self._previous_point is None or (
-            self._instrument_aggregation_temporality
-            is collection_aggregation_temporality
-        ):
-            self._previous_point = current_point
-            return current_point
+                    if current_value is None:
```

Comment:
Style nit, but what do you think about extracting a couple of methods for readability, perhaps like this?

```
            if (self._instrument_aggregation_temporality is AggregationTemporality.DELTA):
                # This happens when the corresponding instrument for this
                # aggregation is synchronous.
                if (collection_aggregation_temporality is AggregationTemporality.DELTA):
                    return self.create_delta(collection_start_nano, current_value, max_, min_, sum_)

                return self.create_cumulative(collection_start_nano, current_value, max_, min_, sum_)
```
The methods could also potentially be independently tested.

Reply:
The problem with that approach is that when reading the code, the reader needs to make a mental copy of the values passed into that function, find the line where the function is defined, mentally paste the values, continue reading, make a mental copy of the returned value, go back to where the reader was, mentally paste the returned value and continue.

If this code is called more than once an additional function is a good idea, but in this case I don't see much use for it.

Reply:
I understand and very much respect your preference and that of the SIG, as well as Python idioms in general, but I think it's often a healthy practice to break functions up into small, focused, well-named ones, even if those smaller ones are only called from one place. Happy to discuss separately.